pkg/resource/resource.go

```diff
@@ -21,18 +21,25 @@ type WalkFunc func(path string, reader io.Reader) error
 type Resource interface {
 	Clone(path string) error
 	ClonePath() string
+	Critical(code logger.ErrorCode, msg string, args ...any)
+	Debug(code logger.ErrorCode, msg string, args ...any)
 	Depth() uint16
 	EnrichResult(result *response.Result) *response.Result
+	Error(code logger.ErrorCode, msg string, args ...any)
 	ID() string
+	Info(code logger.ErrorCode, msg string, args ...any)
 	Kind() string
+	Logs() []logger.Entry
 	Priority() int
 	ReadFile(path string) ([]byte, error)
 	SetCloneTimeout(timeout time.Duration)
 	SetDepth(depth uint16)
+	SetResourceLogs(enabled bool)
```

Comment:
```suggestion
	IncludeLogs(enabled bool)
```

Thoughts? (or something similar)

Rationale:

- SetResourceLogs at first glance felt like it was a setter for a logs field before I got over to the params.